pkg/util/push/otel.go

```diff
@@ -193,11 +198,49 @@ func promToMimirTimeseries(promTs *prompb.TimeSeries) mimirpb.PreallocTimeseries
 	ts := mimirpb.TimeseriesFromPool()
 	ts.Labels = labels
 	ts.Samples = samples
+	ts.Histograms = histograms
 	ts.Exemplars = exemplars
 
 	return mimirpb.PreallocTimeseries{TimeSeries: ts}
 }
 
+func promToMimirHistogram(h *prompb.Histogram) mimirpb.Histogram {
+	pSpans := make([]mimirpb.BucketSpan, len(h.PositiveSpans))
```

Comment:
The preallocation done here and below is incorrect. I opened a PR to fix it: https://github.com/grafana/mimir/pull/4639